models/user.rb

```diff
@@ -25,6 +25,11 @@ def password=(new_password)
     super Argon2::Password.create(new_password)
   end
 
+  def can_reset
+    reset_time = settings['last_password_reset'].to_i + RESET_WINDOW
+    reset_time < Time.now.to_i
```

Comment:
settings['last_password_reset']&.to_i || 0 < Time.now.to_i - RESET_WINDOW

it might be nil right?